earth_enterprise/src/server/wsgi/serve/publish/publish_manager.py

```diff
@@ -114,6 +114,7 @@ def __init__(self):
       raise exceptions.PublishServeException(
           "Couldn't get self-referential server URL.")
     logger.debug("PublishManager init: server URL %s", server_url)
+    os.environ["GESERVER_URL"] = server_url
```

Comment:
Is there a way to do this without modifying the os.environ? There should be a better way to share info within the class.

Reply:
I  thought if we need this information later, outside the class as well , this will be centralized place

Reply:
I don't think it's a good idea to access the OS environment from within a web service. They are meant to be self-contained. We have no real control over how our environment is setup and there's no guarantee for repeatable behavior because we're at the mercy of how the web server wants to behave. 

Looks like this class is already storing member variables. Just a few lines up you have `self._publish_helper = publish_manager_helper.PublishManagerHelper()`. I suggest following that same pattern here.

Reply:
ok, I will add a member variable.